src/System.Slices/System/SpanExtensions_binary.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>
+    /// A collection of convenient span helpers, exposed as extension methods.
+    /// </summary>
+    public static partial class SpanExtensions
+    {
+        /// <summary>
+        /// Casts a Slice of one primitive type (T) to another primitive type (U).
+        /// These types may not contain managed objects, in order to preserve type
+        /// safety.  This is checked statically by a Roslyn analyzer.
+        /// </summary>
+        /// <param name="slice">The source slice, of type T.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Span<U> Cast<[Primitive]T, [Primitive]U>(this Span<T> slice)
```

Comment:
I see ... it just an attribute. I have not realized that you can stick attributes on generic parameters in C#.